application/src/main/java/org/togetherjava/tjbot/features/help/PinnedAnnouncementListener.java

```diff
@@ -0,0 +1,56 @@
+package org.togetherjava.tjbot.features.help;
+
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageType;
+import net.dv8tion.jda.api.entities.channel.concrete.ThreadChannel;
+import net.dv8tion.jda.api.entities.channel.unions.MessageChannelUnion;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.features.MessageReceiverAdapter;
+
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+/**
+ * Listener that receives all pinned announcement messages from helper forum so that it can be
+ * removed
+ */
+public final class PinnedAnnouncementListener extends MessageReceiverAdapter {
```

Comment:
im not a fan of the name. id go for something in the direction of `RemoveHelpThreadPinMessageListener`. Perhaps u can find a shorter name in that direction

Reply:
"RemoveHelpThreadPinMessageListener" sounds like it removes "pinned messages", imo it's more like an announcement for everyone to see something is pinned. Wanted to go with "HelpThreadPinnedAnnouncementListener" but it's a mouthful. Then is it necessary to add "help thread"? It's already in help package so that kinda of gives it away.

Reply:
It doesnt just listen to it. It fulfils a very specific purpose, namely deleting them. If you would create a second feature that acts "on pin notifications", it would be in a separate class. hence why i think its crucial to have the actual action/purpose in the name. and not just the "when"

Reply:
`PinNotificationRemover`?

Reply:
Much better